Closing CDATA tag throws exception

With lsp4xml 0.5.0 from master, given
```xml
<script>
   <![CDATA[
      <message>Foo</message>
   ]]
</script>
```
when you type the closing bracket on L4, an exception is thrown:

```
Message: Internal error: java.lang.ClassCastException: org.eclipse.lsp4xml.dom.DOMCDATASection cannot be cast to org.eclipse.lsp4xml.dom.DOMElement
java.util.concurrent.CompletionException: java.lang.ClassCastException: org.eclipse.lsp4xml.dom.DOMCDATASection cannot be cast to org.eclipse.lsp4xml.dom.DOMElement
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:604)
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:577)
	at java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:443)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.ClassCastException: org.eclipse.lsp4xml.dom.DOMCDATASection cannot be cast to org.eclipse.lsp4xml.dom.DOMElement
	at org.eclipse.lsp4xml.services.XMLCompletions.doTagComplete(XMLCompletions.java:341)
	at org.eclipse.lsp4xml.services.XMLLanguageService.doTagComplete(XMLLanguageService.java:193)
	at org.eclipse.lsp4xml.services.XMLLanguageService.doAutoClose(XMLLanguageService.java:203)
	at org.eclipse.lsp4xml.XMLLanguageServer.lambda$closeTag$1(XMLLanguageServer.java:211)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:602)
	... 6 more

```

There's no such error in lsp4xml 0.4.0

--- a/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLCompletions.java
+++ b/org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLCompletions.java
@@ -338,6 +338,9 @@ public class XMLCompletions {
 		String snippet = null;
 		if (c == '>') { // Case: <a>|
 			DOMNode node = xmlDocument.findNodeBefore(offset);
+			if(!(node instanceof DOMElement)) {
+				return null;
+			}
 			DOMElement element = ((DOMElement) node);
 			if (node != null 
 					&& node.isElement() 

--- a/org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/services/XMLCompletionTest.java
+++ b/org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/services/XMLCompletionTest.java
@@ -156,6 +156,15 @@ public class XMLCompletionTest {
 		testCompletionFor("<a>   <div|    <a>", true, c("div", "<div></div>"));
 	}
 
+	@Test
+	public void testnoCDATANPE() {
+		try {
+			testCompletionFor("<a> <![CDATA[<b>foo</b>]]| </a>", 0);
+		} catch (BadLocationException e) {
+			fail();
+		}
+	}
+
 	// -------------------Tools----------------------------------------------------------
 
 	public void assertOpenStartTagCompletion(String xmlText, int expectedStartTagOffset, boolean startWithTagOpen,

